system/user/addons/wb_category_select/ft.wb_category_select.php

```diff
@@ -596,15 +624,33 @@ private function _add_js_css($field_id, $cell)
 			ee()->session->set_cache('wb_category_select', 'cp_assets_set', true);
 		}
 
-		if ($cell)
+		if ($cell == 'matrix')
 		{
-			ee()->cp->add_to_foot('
-			<script type="text/javascript">
-				(function($) {
-					Matrix.bind("wb_category_select", "display", function(cell){ $(this).find("select").multi(); });	
-				})(jQuery);
-			</script>
-			');
+			if ( ! ee()->session->cache('wb_category_select', 'cp_matrix_set'))
+			{
+				ee()->cp->add_to_foot('
+				<script type="text/javascript">
+					(function($) {
+						Matrix.bind("wb_category_select", "display", function(cell){ $(this).find("select").multi(); });	
+					})(jQuery);
+				</script>
+				');
+				ee()->session->set_cache('wb_category_select', 'cp_matrix_set', true);
+			}
+		}
+		elseif ($cell == 'grid')
+		{
+			if ( ! ee()->session->cache('wb_category_select', 'cp_grid_set'))
```

Comment:
Same as above:
```suggestion
		elseif ($cell == 'grid' && ! ee()->session->cache('wb_category_select', 'cp_grid_set'))
		{
```

Reply:
This too would cause the last condition being met unnecessarily. I could've tidied this all up more though I didn't want to move away from your current code too much. I'm happy to change however you suggest though. Thanks